xdsl/dialects/csl.py

```diff
@@ -1486,6 +1533,13 @@ class ParamOp(IRDLOperation):
 
     res = result_def(T)
 
+    def __init__(self, name: str, result_type: T):
```

Comment:
I'm not sure if this is allowed in python... @PapyChacal do you have any input on this? I think you know more on this than I do...

Reply:
I thought that's what make TypeVars so neat
https://mypy.readthedocs.io/en/stable/generics.html

Reply:
I don't think it's necessarily wrong, but it's not expressing what you may think it is.

Reply:
I misread, it's not a TypeVar. I think it's not an IRDL constraint either, though, it should just be a union type with some additional meta data.